xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -3562,6 +3562,24 @@ bool CVideoPlayer::OpenVideoStream(CDVDStreamInfo& hint, bool reset)
   if (hint.stereo_mode.empty())
     hint.stereo_mode = CStereoscopicsManager::GetInstance().DetectStereoModeByString(m_item.GetPath());
 
+  SelectionStream& s = m_SelectionStreams.Get(STREAM_VIDEO, 0);
```

Comment:
shouldn't the hint also be updated according to videoSettings? And this section looks somewhat familiar, but I don't recall where it has been used. Only mentioned this so that the logic could probably been moved to some utility class if used more than once. OTOH it's not a really complex logic.